### PR TITLE
[cost-insights] Export alerts utils

### PR DESCRIPTION
**Pull request by brendasukh**
## Hey, I just made a Pull Request!
- Fixes regression where alerts classes are no longer exposed due to restructuring of utils/types files
- Updates `getProductInsights` in the API to expect a `lastBillingDate` param  as a follow up to https://github.com/backstage/backstage/pull/3054, to account for billing data delays

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

File: plugins/cost-insights/src/api/CostInsightsApi.ts [hunk 1]
```diff
@@ -112,12 +112,14 @@ export type CostInsightsApi = {
...
    * of how the durations are interpreted in Cost Insights.
+   * @param lastBillingDate The most current date for which billing data is complete, in YYYY-MM-DD format.
    * @param project (optional) The project id from getGroupProjects or query parameters
    */
   getProductInsights(
     product: string,
     group: string,
     duration: Duration,
+    lastBillingDate: string,
     project: Maybe<string>,
```
Comment (ryanvazquez) [inline plugins/cost-insights/src/api/CostInsightsApi.ts:123]: Params for this method are starting to get a bit verbose :/

File: plugins/cost-insights/src/index.ts [hunk 2]
```diff
@@ -18,7 +18,8 @@ export { plugin } from './plugin';
 export * from './client';
 export * from './api';
 export * from './components';
-export { useCurrency } from './hooks';
+export { useCurrency, CurrencyContext } from './hooks';
```
Comment (ryanvazquez) [inline plugins/cost-insights/src/index.ts:21]: What do we need to export `CurrencyContext` for?

File: plugins/cost-insights/src/index.ts [hunk 3]
```diff
@@ -18,7 +18,8 @@ export { plugin } from './plugin';
...
 export * from './types';
 export * from './utils/tests';
 export * from './utils/duration';
+export * from './utils/alerts';
```
Comment (ryanvazquez) [inline plugins/cost-insights/src/index.ts:25]: 😞

File: plugins/cost-insights/src/api/CostInsightsApi.ts (hunk 1 above)
Comment (freben) [reply plugins/cost-insights/src/api/CostInsightsApi.ts:123]: In general with more than like 1-2 parameters, we have started favoring passing in a single object with the corresponding fields instead - long lists of unnamed (at the call site) params are a huge pain

File: plugins/cost-insights/src/index.ts (hunk 2 above)
Comment (brendasukh) [reply plugins/cost-insights/src/index.ts:21]: @ryanmvazquez We need it to mock the currency provider when using `CostGrowth` internally in GKE alerts.

Comment (ryanvazquez) [reply plugins/cost-insights/src/index.ts:21]: Just for testing? We already export `MockCurrencyContext` in `./utils/tests`

File: plugins/cost-insights/src/index.ts (hunk 3 above)
Comment (ryanvazquez) [reply plugins/cost-insights/src/index.ts:25]: Also, I think we need to figure out a more consistent approach for handling public exports at some point in the near future. At the moment, we have about three different approaches - wholesale export from index (types), partial export from index (components), and export from file (utils).

File: .changeset/rich-dodos-smell.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
```
Comment (timbonicus) [inline .changeset/rich-dodos-smell.md:1]: We have codeowners set up for changesets if you manually tweak the filename to `cost-insights-X.md`

File: plugins/cost-insights/src/index.ts (hunk 3 above)
Comment (Rugvip) [reply plugins/cost-insights/src/index.ts:25]: Use this: https://backstage.io/docs/architecture-decisions/adrs-adr004 ! 😁

**Review by ryanvazquez — APPROVED** [review]
LGTM :shipit:

**Pull request merged**